benchmarks/hash_anchored_vs_plain.py

```
@@ -0,0 +1,427 @@
+    avg_batch = sum(batch_times) / len(batch_times) * 1000
+    speedup = avg_individual / avg_batch if avg_batch > 0 else 0
+
+    print(f"  Iterations:          {iterations}")
+    print(f"  Edits per iteration: {n_edits}")
+    print()
+    print(f"  Individual (N calls): {avg_individual:.1f} ms total")
+    print(f"  Batch (1 call):       {avg_batch:.1f} ms total")
+    print(f"  Speedup:              {speedup:.1f}x")
+    print()
+    print(f"  Individual: {n_edits} x list_paragraphs() + {n_edits} x replace()")
+    print(f"  Batch:      1 x list_paragraphs() + 1 x batch_edit()")
```

Ruff (0.15.7)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)